tests/core/pyspec/eth2spec/test/eip7732/block_processing/test_process_execution_payload_header.py

```diff
@@ -0,0 +1,463 @@
+    else:
+        expect_assertion_error = False
+
+    if expect_assertion_error:
+        try:
+            spec.process_execution_payload_header(state, block)
+            assert False, "Expected AssertionError but none was raised"
+        except AssertionError:
+            pass
+        yield "post", None
+        return
+
+    spec.process_execution_payload_header(state, block)
+    yield "post", state
+
+
+def prepare_signed_execution_payload_header(
+    spec,
+    state,
+    builder_index=None,
+    value=None,
+    slot=None,
+    parent_block_hash=None,
+    parent_block_root=None,
+    fee_recipient=None,
+    gas_limit=None,
+    block_hash=None,
+    blob_kzg_commitments_root=None,
+    valid_signature=True,
+):
+    """
+    Helper to create a signed execution payload header with customizable parameters.
+    Note: Always call build_empty_block_for_next_slot BEFORE this function if using it,
+    since that function advances the state slot.
+    """
+    if builder_index is None:
+        builder_index = spec.get_beacon_proposer_index(state)
+
+    if slot is None:
+        slot = state.slot
+
+    if parent_block_hash is None:
+        parent_block_hash = state.latest_block_hash
+
+    if parent_block_root is None:
+        parent_block_root = state.latest_block_header.hash_tree_root()
+
+    if fee_recipient is None:
+        fee_recipient = spec.ExecutionAddress()
+
+    if gas_limit is None:
+        gas_limit = spec.uint64(30000000)
+
+    if block_hash is None:
+        block_hash = spec.Hash32()
+
+    if value is None:
+        value = spec.Gwei(0)
+
+    if blob_kzg_commitments_root is None:
+        kzg_list = spec.List[spec.KZGCommitment, spec.MAX_BLOB_COMMITMENTS_PER_BLOCK]()
+        blob_kzg_commitments_root = kzg_list.hash_tree_root()
+
+    header = spec.ExecutionPayloadHeader(
+        parent_block_hash=parent_block_hash,
+        parent_block_root=parent_block_root,
+        block_hash=block_hash,
+        fee_recipient=fee_recipient,
+        gas_limit=gas_limit,
+        builder_index=builder_index,
+        slot=slot,
+        value=value,
+        blob_kzg_commitments_root=blob_kzg_commitments_root,
+    )
+
+    if valid_signature:
+        privkey = privkeys[builder_index]
+        signature = spec.get_execution_payload_header_signature(state, header, privkey)
+    else:
+        # Invalid signature
+        signature = spec.BLSSignature()
+
+    return spec.SignedExecutionPayloadHeader(
+        message=header,
+        signature=signature,
+    )
+
+
+def make_validator_builder(spec, state, validator_index):
+    """
+    Helper to make a validator a builder by setting builder withdrawal credentials
+    """
+    # Set builder withdrawal credentials (0x03 prefix)
+    builder_credentials = spec.BLS_WITHDRAWAL_PREFIX + b"\x00" * 31
+    builder_credentials = spec.BUILDER_WITHDRAWAL_PREFIX + builder_credentials[1:]
+    state.validators[validator_index].withdrawal_credentials = builder_credentials
```

Comment:
For consistency, we should probably mirror these functions:

https://github.com/ethereum/consensus-specs/blob/56899c5dbb8511c4fe10ec05d0b2afd1159882fb/tests/core/pyspec/eth2spec/test/helpers/withdrawals.py#L127-L151

Reply:
modified in [4f2db69](https://github.com/ethereum/consensus-specs/pull/4457/commits/4f2db6950780bf31f4b53c922d0c7f8c6ce9053c)